src/AppBundle/Service/ShiftService.php

```diff
@@ -48,7 +50,41 @@ public function __construct(EntityManagerInterface $em, BeneficiaryService $bene
     public function remainingToBook(Membership $member)
     {
         $cycle_end = $this->membershipService->getEndOfCycle($member);
-        return $this->due_duration_by_cycle - $member->getTimeCount($cycle_end);
+
+        $remaining = $this->due_duration_by_cycle - $member->getTimeCount($cycle_end);
+
+        // also take into account planned shifts
+        if ($this->use_card_reader_to_validate_shifts) {
```

Comment:
Pourquoi mettre ce `if` ici ?

J'aurais plutot vu la condition : 

true si le compteur est négatif OU pas 3h de créneau sur le cycle courant

Reply:
Ca permettrait d'être indépendant du code de validation des créneaux.

Reply:
Niveau implémentation, il faudrait regarder les futurs créneaux non validés.

(partant du principe auqnd dans le cas not `use_card_reader_to_validate_shifts`, les créneaux sont validés par défaut)

Reply:
je peux enlever le if, mais c'était pour éviter de faire des requêtes DB pour les autres coop qui n'utilisent pas ce système, et donc qui ne peuvent pas avoir de créneau à venir qui n'apparaisse pas encore dans le compteur temps.

Si je comprends bien, tu proposes d'enlever l'appel à `getTimeCount`, et regarder plutôt la durée total des créneaux du cycle en cours ?